### DISCUSSION TITLE
I require violentmonkey code for this AHK script

### DISCUSSION THREAD

**Discussion by vsrawat** [violentmonkey/violentmonkey#1497] [Q&A]
`Space:: Send, {Ctrl Down}

*Space::
Send, {Ctrl Up}
return
ExitApp`

The above is a tiny Autohotkey script. It just makes spacebar toggle "Control" key. Pressing Space once makes Control key get pressed and remain so. Pressing Space again makes Control key get released. 

As AHK extension is not available for Vivaldi (Chrome) Browser, I need to convert it to some other coding method.

Can someone please give me equivalent code so that same work gets happen using Violentmonkey.

I am a newbie, I had picked a script from AHK forum and converted to the above for my need. If there is any redundancy or something, I am not aware. You can modify it for that.

Thanks.

**Comment by phil294** [comment]
Why can't you use this ahk script in Vivaldi? AutoHotkey works independently from the underlying program.

Violentmonkey is to modify web pages using JS. To get started with JavaScript, check out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide. Hotkeys work with Event listeners in JS, e.g. https://developer.mozilla.org/en-US/docs/Web/API/Document/keydown_event. Keeping a key pressed is not possible with JS afaik, but I also have no idea what you are trying to do

**Reply by vsrawat** [reply]
Thanks for replying.

Using AHK globally on pc would keep the above behavior pc-wide. That creates problem when I unknowing switch to other work on pc, and forget to suspend or exit AHK.

So, it would be convenient if such a keymapping behavior works only under Vivaldi on a particular page of a particular website. That way I don't have to keep on altering setting many times on entire pc when I want this to be off/on.

So, in absence of AHK extension for Vivaldi (Chrome), a Vivaldi-ONLY method would be simpler for me. Violentmonkey is one script-manager that I am already using, so converting the above to vm script would help me.

Thanks.
